gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/common/impl/SimpleConfigStoreResolver.java

```diff
@@ -0,0 +1,151 @@
+ * @author mitu
+ *
+ */
+public class SimpleConfigStoreResolver implements ConfigStoreWithResolution {
+
+  private static final Logger LOG = Logger.getLogger(SimpleConfigStoreResolver.class);
+  private final ConfigStore store;
+
+  public SimpleConfigStoreResolver(ConfigStore base) {
+    this.store = base;
+  }
+
+  @Override
+  public String getCurrentVersion() {
+    return store.getCurrentVersion();
+  }
+
+  @Override
+  public URI getStoreURI() {
+    return store.getStoreURI();
+  }
+
+  @Override
+  public Collection<URI> getChildren(URI uri, String version) throws VersionDoesNotExistException{
+    return store.getChildren(uri, version);
+  }
+
+  @Override
+  public List<URI> getOwnImports(URI uri, String version) throws VersionDoesNotExistException{
+    return store.getOwnImports(uri, version);
+  }
+
+  @Override
+  public Config getOwnConfig(URI uri, String version) throws VersionDoesNotExistException{
+    return store.getOwnConfig(uri, version);
+  }
+
+  @Override
+  public Config getResolvedConfig(URI uri, String version) throws VersionDoesNotExistException{
+    UUID traceID = UUID.randomUUID();
+    LOG.info(String.format("Getting resolved config for uri: %s for version %s, with traceID: %s", 
+        uri, version, traceID));
+    return getResolvedConfigWithTrace(uri, version, traceID.toString());
+  }
+
+  protected Config getResolvedConfigWithTrace(URI uri, String version, String traceID) throws VersionDoesNotExistException{
+    CircularDependencyChecker.checkCircularDependency(store, version, uri);
+    Config self = this.getOwnConfig(uri, version);
+    for (Entry<String, ConfigValue> entry : self.entrySet()) {
+      LOG.info(String.format("key %s, value %s from uri %s, with traceID %s", 
+          entry.getKey(), entry.getValue(), uri, traceID)); 
+    }
+
+    // root can not include anything, otherwise will have circular dependency 
+    if (isRootURI(uri)) {
+      return self;
+    }
+
+    // The order is already decided in imported List : higher priority appear in the head
+    List<URI> imported = this.getOwnImports(uri, version);
+    for (URI u : imported) {
+      Config singleImported = this.getResolvedConfigWithTrace(u, version, traceID);
+      self = self.withFallback(singleImported);
+    }
+
+    // all the imports have higher priority than ancestor
+    Config ancestor = this.getAncestorConfig(uri, version, traceID);
+    return self.withFallback(ancestor);
+  }
+
+  protected Config getAncestorConfig(URI uri, String version, String traceID) throws VersionDoesNotExistException{
+    URI parent = PathUtils.getParentURI(uri);
+    Config res = getResolvedConfigWithTrace(parent, version, traceID);
+    return res;
+  }
+
+  protected static final boolean isRootURI(URI uri) {
+    if (uri == null)
+      return false;
+
+    return uri.toString().length() == 0;
+  }
+
+  @Override
+  public List<URI> getImportsRecursively(URI uri, String version) throws VersionDoesNotExistException{
```

Comment:
Should this method also return implicit imports coming from the ancestors including the ancestors themselves? Can you remind me what the intended purpose of this method is?